### DISCUSSION TITLE
Have a simple landing page for the project (GitHub Pages?)

### DISCUSSION THREAD

**Discussion by tooomm** [paperless-ngx/paperless-ngx#976] [Feature Requests] [answered]
I would like to see Paperless having a basic web presence (besides its documentation) targeting new users and presenting the project and its core benefits. When searching about the "paperless office" topic or about this tool in particular, people only find this repo or the docs as official pages. Both can be overwhelming at first and are not the best initial contact point to provide certain information at a glance. They serve different purposes.

Projects having a small web presence just shine in a brighter light right away and users find what they are looking for quicker and easier.
Probably a good fit with the new org launch and project rebirth. 🚀 

There is no need for a huge, complex webpage. Some information that can be helpful for interested people:
(What are they most likely looking for? Which questions do they search answers for? Where do they need help? ...)
- Describe key features
- Showcase screenshots
- Provide download options (GitHub Releases, DockerHub...)
- Link to the docs
- Link to GitHub repo, CrowdIn...

Additional ideas:
- Link to important guides, tutorials or help to get users started
- Have a FAQ section
- ...

Ideally, it should be set up once and does not require much further maintenance as a general project page. It can also be combined with the docs page for a integrated experience.
There are many free, responsive themes available for various static site generators that process e.g. md files (like **Sphinx** which is used for the docs currently):
 - **Jekyll** (Ruby/Liquid) - Project Page
   - "[Serif](https://jekyllthemes.io/theme/serif)" is also available for Hugo
   - ⭐ "[Creative](https://jamstackthemes.dev/theme/jekyll-creative-theme-new)" and "[Agency](https://jekyll-themes.com/agency-jekyll-theme)" are similar
   - ⭐ "[New Age](https://jamstackthemes.dev/theme/jekyll-new-age)"
   - "[Ubuild](https://jamstackthemes.dev/theme/ubuild-jekyll)"
   - "[Helium](https://jamstackthemes.dev/theme/jekyll-helium-theme)"
   - "[Cause](https://jamstackthemes.dev/theme/jekyll-cause)"

 - **MkDocs** (Python/Jinja2 - same as Sphinx) - Project Page & Docs
   - ⭐ "[Materials](https://jamstackthemes.dev/theme/mkdocs-material/)" greatly shows how an informative landing page can be combined with docs

 - **VuePress** (Javascript/Vue) - Project Page & Docs
   - "[Default](https://v2.vuepress.vuejs.org)" theme is already pretty and modern
   - ⭐ "[Hope](https://vuepress-theme-hope.github.io/v2)"
   - ⭐ "[Mix](https://vuepress-theme-mix.vercel.app)"

 - **Hugo** (Go) - Project Page
   - ⭐ "[Doks](https://jamstackthemes.dev/theme/hugo-doks)" also allows combining a project page & docs with Hugo
   - "[coHub](https://jamstackthemes.dev/theme/hugo-cohub)"
   - ⭐ "[Apsho](https://jamstackthemes.dev/theme/hugo-apsho)"
   - "[Portio](https://github.com/StaticMania/portio-hugo)"
   - "[Landing Page](https://jamstackthemes.dev/theme/hugo-landing-page)"
   - "[Serif](https://jamstackthemes.dev/theme/hugo-serif)" is also available for Jekyll
   - "[Webslides](https://jamstackthemes.dev/theme/hugo-webslides)" is something a little different...
   - "[Hero](https://jamstackthemes.dev/theme/hugo-hero-theme)"

 - **Docusaurus** (Javascript/React, md) or [others](https://jamstack.org/generators)... they vary on complexity, features and customization or plugins.

 - There is also https://github.com/files-community/website/ as a really nice example that combines project information and docs (It uses [SvelteKit](https://svelte.dev/))

I could not find a base theme for Sphinx to accomplish something like a little project page with some general information out of the box - always docs only. That's why only other tools/themes are listed.
Selecting the right tool is dependent on what the site should be like or develop into - like is a combined project landing page with docs a (future) option? Or the general info page should be separate and a more sophisticated style is wanted?
Of course it matters what the people working with it prefer and that it's not to complex to deal with. For most tools, Markdown is the default content source.

The result can be hosted at e.g. [GitHub Pages](https://pages.github.com/) (`https://paperless-ngx.github.io` or `/docs`) without ads and for free.
I just learned that the project (shamoon?) owns the https://paperless-ngx.com domain. GitHub Pages supports custom domains as well. 

The goal of this discussion thread is to collect first opinions and ideas about the topic.
What do you think? Which approach or theme do you like? Feel free to post your ideas and opinion!

**Comment by pascaliske** [comment]
Just to add: MkDocs, as you already mentioned, is great in combination with Material and GitHub Actions. Just have a look at the following two links:

- https://squidfunk.github.io/mkdocs-material/
- https://github.com/mhausenblas/mkdocs-deploy-gh-pages

Those two combined provide a fast, easy and painless way to build a landing page based on Markdown. Just create a workflow which takes some markdown files and publishes a ready-to-deploy site in the `gh-pages` branch which then can be served by GitHub Pages (optionally under a custom domain).

**Reply by tooomm** [reply]
Thanks for the links and informations. The materials theme you linked is actually the same I have in the list above - a very nice theme that is one of my favourites to adopt for Paperless-ngx!

Since all mentioned tools are static sites generators and GitHub Pages supports hosting static sites, they can all be combined with GH Actions and should work well in an automated setup where updates to site content gets deployed right away without any other external services.

**Comment by shamoon** [comment, marked as answer]
Hey all, thanks for your support of this project. Take a peek at https://github.com/paperless-ngx/paperless-ngx/pull/2067 if youre interested

**Reply by tooomm** [reply]
Wow, on a first quick look it's already very very nice. The mkdocs theme is a great choice! 🤩 
I'm low on free time currently and might not be able to check the PR closer right now or provide more detailed feedback though.
